azurecompute/src/main/java/org/jclouds/azurecompute/util/ConflictManagementPredicate.java

```diff
@@ -192,17 +196,19 @@ public OperationSucceededPredicate(final AzureComputeApi api) {
       @Override
       public boolean apply(final String input) {
          final Operation operation = api.getOperationApi().get(input);
+         if (operation == null) return false;
          switch (operation.status()) {
             case SUCCEEDED:
                return true;
-
             case IN_PROGRESS:
             case UNRECOGNIZED:
                return false;
-
             case FAILED:
-               throw new RuntimeException(new CancellationException(input));
-
+               String message = String.format("Operation %s failed for unknown reasons", input);
+               if (operation.error() != null) {
+                 message = operation.error().message();
```

Comment:
I wonder if the toString on `operation`, or on `operation.error()`, would give us more information than the message (e.g. it might include an error code etc).